project/tests/com/google/daggerquery/executor/models/QueryTest.java

```diff
@@ -125,12 +218,57 @@ public void testExecutingDepsQuery_WithNullBindingGraph_ThrowsNullPointerExcepti
     BindingGraphProto.BindingGraph.ListWithDependencies factoryNodeDeps = BindingGraphProto.BindingGraph.ListWithDependencies.newBuilder()
         .addDependency(catNode)
         .build();
-    BindingGraphProto.BindingGraph.ListWithDependencies catNodeDeps = BindingGraphProto.BindingGraph.ListWithDependencies.newBuilder().build();
+    BindingGraphProto.BindingGraph.ListWithDependencies catNodeDeps = BindingGraphProto.BindingGraph.ListWithDependencies.newBuilder()
+        .build();
+    BindingGraphProto.BindingGraph.ListWithDependencies helperNodeDeps = BindingGraphProto.BindingGraph.ListWithDependencies.newBuilder()
+        .build();
+
+    return BindingGraphProto.BindingGraph.newBuilder()
+        .putAdjacencyList("com.google.Component", componentNodeDeps)
+        .putAdjacencyList("com.google.CatsFactory", factoryNodeDeps)
+        .putAdjacencyList("com.google.Cat", catNodeDeps)
+        .putAdjacencyList("com.google.Helper", helperNodeDeps)
+        .build();
+  }
+
+  /*
+   * Makes an acyclic binding graph with multiple paths between nodes with the following structure:
+   *
+   * com.google.Component --> com.google.CatsFactory --> com.google.Cat --> com.google.Details
+   * com.google.Component --> com.google.Helper --> com.google.CatsFactory --> com.google.Cat --> com.google.Details
+   * com.google.Component --> com.google.Cat --> com.google.Details
+   * com.google.Component --> com.google.Details
```

Comment:
This is great. Having the graph described in the comments makes this code much easier to understand.